build/docs.js

```diff
@@ -29,18 +33,47 @@ function orderObject (obj) {
  * Ensures that the repo cloneable at the provided address is downloaded
  * and up-to-date.
  * @param  {String} addr
- * @param  {Function} callback
+ * @return {Promise}
  */
-function getRepo (addr, callback) {
+function getRepo (addr) {
     // extract everything after the last slash of the path, excluding .git:
     const name = (/\/([^/]*?)(\.git)?$/).exec(addr)[1];
     const target = path.join(config.src.tmp, name);
-    fs.lstat(target, (err, stats) => {
-        if (!err && stats.isDirectory()) {
-            childProcess.exec(`cd ${target} && git pull`, callback);
-        } else {
-            childProcess.exec(`cd ${config.src.tmp} && git clone ${addr}`, callback);
+    const statPromise = fs.lstatAsync(target);
+    // We have a repo override
+    if (config.repos && config.repos[name]) {
+        return statPromise.then(stats => {
+            // If it's a cloned copy, ripe it
+            if (!stats.isSymbolicLink() && stats.isDirectory()) {
+                return del(target);
+            }
+        })
+        // if it didn't exist, ignore.
+        .catch({ code: 'ENOENT' }, () => { /* do nothing */ })
+        .then(() => fs.symlinkAsync(path.join(__dirname, '../', config.repos[name]), target))
+        // if it existed already, ignore it.
+        .catch({ code: 'EEXIST' }, () => { /* do nothing */ });
+    }
+
+    return statPromise.tap(stats => {
+        // Delete symbolic if present
+        if (stats.isSymbolicLink()) {
+            return fs.unlinkAsync(target);
         }
+    })
+    .tap(stats => {
```

Comment:
There's not a need for this, strictly speaking; if it's a symb link it's not a directory